service/controller/v2/resource/deployment/resource.go

```diff
@@ -97,16 +100,39 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			},
 		},
 	}
-	f, err := deploymentsClient.CreateOrUpdate(ctx, resourceGroupName, mainDeployment.Name, d)
-	if err != nil {
+
+	d, err := deploymentsClient.Get(ctx, resourceGroupName, mainDeployment.Name)
+	if IsNotFound(err) {
+		_, err := deploymentsClient.CreateOrUpdate(ctx, resourceGroupName, mainDeployment.Name, newDeployment)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring deployment initiated")
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation for custom object")
+
+		return nil
```

Comment:
Do we need that `if` content? Can't we just fall trough?

Reply:
The case here is that we do not have a deployment. So this marks the very first reconciliation of it. That means we want to trigger the creation and cancel the reconciliation completely. Otherwise following resources fail because of missing Azure resources down the road. When we fall through we trigger creation but we would need to do some other weird stuff for the cancelation. It wouldn't get better I guess. Ideas welcome though.

Reply:
We need to fix other resources so they can handle that. There is not guarantee those will be created on next reconciliation loop.

Reply:
https://github.com/giantswarm/giantswarm/issues/3286

Reply:
I will check on this.